jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ConnectionMetaData.java

```diff
@@ -77,10 +78,11 @@ static HostPort getServerAuthority(HttpConfiguration httpConfiguration, Connecti
         if (authority != null)
             return authority;
 
-        SocketAddress local = connectionMetaData.getLocalSocketAddress();
-        if (local instanceof InetSocketAddress inet)
-            return new HostPort(inet.getHostString(), inet.getPort());
-
+        SocketAddress localSocketAddress = connectionMetaData.getLocalSocketAddress();
+        if (localSocketAddress instanceof InetSocketAddress inetSocketAddress)
+            return new HostPort(inetSocketAddress.getHostString(), inetSocketAddress.getPort());
+        else if (localSocketAddress != null)
+            return new HostPort(localSocketAddress.toString(), connectionMetaData.isSecure() ? 443 : 80);
```

Comment:
Is this to support Unix-Domain?
If so, I am not sure we want to add an inet port to a non-inet host.

Also, method needs javadocs.

Reply:
@sbordet so return null or a HostPort with no port?

Reply:
I would return null, the idea is that it is a last resort and most of the time there is an authority from somewhere else than the network.